extensions/interactions/customization-args-defs.ts

```diff
@@ -18,25 +18,51 @@
 
 import { IGraphAnswer } from 'interactions/answer-defs';
 
+import { SubtitledHtmlBackendDict, SubtitledHtml } from
+  'domain/exploration/SubtitledHtmlObjectFactory';
+import { SubtitledUnicodeBackendDict, SubtitledUnicode } from
+  'domain/exploration/SubtitledUnicodeObjectFactory';
+
 interface ILabeledRegion {
   region: {
     area: number[][];
   };
   label: string;
 }
 
-interface IImageWithRegions {
+export interface IImageWithRegions {
   labeledRegions: ILabeledRegion[];
   imagePath: string;
 }
 
-interface IReadableMusicNote {
+export interface IReadableMusicNote {
   readableNoteName: string;
 }
 
-export interface IAlgebraicExpressionInputCustomizationArgs { }
 
-export interface ICodeReplCustomizationArgs {
+interface AlgebraicExpressionInputCustomizationArgsBackendDict { }
+export interface AlgebraicExpressionInputCustomizationArgs { }
+type AlgebraicExpressionInputCustomizationArgsBackendDictValue = (
+  AlgebraicExpressionInputCustomizationArgsBackendDict[
+    keyof AlgebraicExpressionInputCustomizationArgsBackendDict]['value']);
+type AlgebraicExpressionInputCustomizationArgsValue = (
+  AlgebraicExpressionInputCustomizationArgs[
+    keyof AlgebraicExpressionInputCustomizationArgs]['value']);
+interface CodeReplCustomizationArgsBackendDict {
+  language?: {
```

Comment:
Why are these optional? With the changes you've made to the backend, shouldn't the keys be deterministic? Ditto in many places below.

/cc @nishantwrp

Reply:
Flagging this for follow-up with @nishantwrp.

Reply:
Hi @shavavo,

From which places in this file have you removed the `?` to get this error?

Reply:
Most of these are related to spec files that don't have those properties. I've created a PR removing `?` that are not needed. (#10093)

Reply:
Hi @shavavo -- @nishantwrp has removed most of the question marks, from what I understand (#10093). Are they still needed here?

Reply:
Yup!